File: KosmicTask/KosmicTask/source/mother framework/KosmicTaskController.php
<?php
	
require_once "spyc.php";
	
class KosmicTaskController {
		
	/*
		 
	 objectToString
		 
	*/
	public static function objectToString($object) {
			
		// get native object as YAML
		$result = Spyc::YAMLDump($object);
			
		return $result;
	}
		
	/*
		 
	 printObject
		 
	 */
	public static function printObject($object) {
			
		// get native object as YAML
		$result = self::objectToString($object);
			
		print $result;
	}
		
}

File: KosmicTask/KosmicTask/source/mother framework/KosmicTaskController.lua
-- http://lua-users.org/wiki/ModulesTutorial

module(..., package.seeall);

-- load yaml shared library
-- this requires a full path so we obtain it from the environment
local path = assert(os.getenv("LUA_YAML_LIB_PATH"))
local yamlLib = assert(package.loadlib(path, "luaopen_yaml"))

-- initialise
yamlLib()

function objectToString(resultObject)
	local result = yaml.dump(resultObject)
	
	local start = "---"
	if string.sub(result,1,string.len(start)) ~= start then
		result = "---\n" .. result
	end
	
	return result
end

function printObject(resultObject)
	local result = objectToString(resultObject)
	print(result)
end